Source/Core/VideoCommon/VideoUtils.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace X11Utils
+{
+class XRRConfiguration;
+}
+
+namespace VideoUtils
+{
+std::vector<std::string> GetAvailableResolutions(X11Utils::XRRConfiguration* xrr_config);
```

Comment:
- If there's no function definition when `APPLE` is defined, there probably shouldn't be a declaration either.
- This function should be in UICommon, not VideoCommon. Semantically, it has nothing to do with the video backends, and it's currently creating a cycle between the two modules (UICommon already depends on VideoCommon/VideoBackendBase).
- This shouldn't take an `XRRConfiguration` pointer when built without XRandR support. You could do something like:
  ```c++
  #ifdef _WIN32
    std::vector<std::string> GetAvailableResolutions();
  #elif defined(HAVE_XRANDR) && HAVE_XRANDR
    std::vector<std::string> GetAvailableResolutions(X11Utils::XRRConfiguration* xrr_config);
  #endif
  ```
  Alternatively, there's probably a way to have it take an opaque native window pointer (in WxWidgets, the result of `GetHandle()`) and internally figure out how to get an XRRConfiguration from that.